ion/cmp_test.go

```diff
@@ -0,0 +1,145 @@
+package ion
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+type ionEqual interface {
+	eq(other ionEqual) bool
+}
+
+type ionFloat struct{ float64 }
+type ionDecimal struct{ *Decimal }
+type ionTimestamp struct{ time.Time }
+
+func (thisFloat ionFloat) eq(other ionEqual) bool {
+	return cmp.Equal(thisFloat.float64, other.(ionFloat).float64, cmpopts.EquateNaNs())
+}
+
+func (thisDecimal ionDecimal) eq(other ionEqual) bool {
+	if val, ok := other.(ionDecimal); ok {
+		if thisDecimal.scale != val.scale {
+			return false
+		}
+		return thisDecimal.Decimal.Equal(val.Decimal)
+	}
+	return false
+}
+
+func (thisTimestamp ionTimestamp) eq(other ionEqual) bool {
+	if val, ok := other.(ionTimestamp); ok {
+		return thisTimestamp.Time.Equal(val.Time)
+	}
+	return false
+}
+
+func cmpAnnotations(thisAnnotations, otherAnnotations []string) bool {
```

Comment:
Since this is comparing `string` slices, I think you might be able to `return DeepEquals(thisAnnotations, otherAnnotations)`?

Reply:
Done